s4/clarity/lims.py

```diff
@@ -91,58 +107,58 @@ def __init__(self, root_uri, username, password, dry_run=False, insecure=False,
 
-        self.container_types = ElementFactory(self, ContainerType, batch_flags=BatchFlags.QUERY)
+        self.container_types = ElementFactory(self, ContainerType)
 
-        self.projects = ElementFactory(self, Project, batch_flags=BatchFlags.QUERY)
+        self.projects = ElementFactory(self, Project)
 
         self.control_types = ElementFactory(self, ControlType)
 
         self.queues = ElementFactory(self, Queue)
 
-        self.reagent_lots = ElementFactory(self, ReagentLot, batch_flags=BatchFlags.QUERY)
+        self.reagent_lots = ElementFactory(self, ReagentLot)
 
-        self.reagent_kits = ElementFactory(self, ReagentKit, batch_flags=BatchFlags.QUERY)
+        self.reagent_kits = ElementFactory(self, ReagentKit)
 
-        self.reagent_types = ElementFactory(self, ReagentType, batch_flags=BatchFlags.QUERY)
+        self.reagent_types = ElementFactory(self, ReagentType)
 
-        self.researchers = ElementFactory(self, Researcher, batch_flags=BatchFlags.QUERY)
+        self.researchers = ElementFactory(self, Researcher)
 
-        self.labs = ElementFactory(self, Lab, batch_flags=BatchFlags.QUERY)
+        self.labs = ElementFactory(self, Lab)
 
-        self.roles = ElementFactory(self, Role, batch_flags=BatchFlags.QUERY)
+        self.roles = ElementFactory(self, Role)
 
-        self.permissions = ElementFactory(self, Permission, batch_flags=BatchFlags.QUERY)
+        self.permissions = ElementFactory(self, Permission)
 
         # configuration
         from .configuration import Workflow, Protocol, ProcessType, Udf, ProcessTemplate, Automation
 
-        self.workflows = ElementFactory(self, Workflow, batch_flags=BatchFlags.QUERY,
-                                        request_path='/configuration/workflows')
-        self.protocols = ElementFactory(self, Protocol, batch_flags=BatchFlags.QUERY,
-                                        request_path='/configuration/protocols')
-        self.udfs = UdfFactory(self, Udf, batch_flags=BatchFlags.QUERY,
-                               request_path='/configuration/udfs')
-        self.process_types = ElementFactory(self, ProcessType, batch_flags=BatchFlags.QUERY,
-                                            name_attribute="displayname")
-        self.process_templates = ElementFactory(self, ProcessTemplate, batch_flags=BatchFlags.QUERY,
-                                                name_attribute="name")
-        self.automations = ElementFactory(self, Automation, batch_flags=BatchFlags.QUERY,
-                                          name_attribute="name", request_path="/configuration/automations")
+        self.workflows = ElementFactory(self, Workflow)
+
+        self.protocols = ElementFactory(self, Protocol)
+
+        self.udfs = UdfFactory(self, Udf)
+
+        self.process_types = ElementFactory(self, ProcessType)
+
+        self.process_templates = ElementFactory(self, ProcessTemplate)
+
+        self.automations = ElementFactory(self, Automation)
```

Comment:
All these factory initializations are so much cleaner now!

Since each property is initialized the exact same way now, I wonder if there is a way to loop through a list of classes and add the properties dynamically, based on the class name? Probably less readable though...

Reply:
Yeah, this is a good description of the services that the LIMS object provides.